src/App.tsx

```diff
@@ -14,11 +14,148 @@ const goodsFromServer: string[] = [
+  state = {
+    goods: goodsFromServer,
+    count: 0,
+    isReversed: false,
+    sortBy: '',
+    lengths: [1, 2, 3, 4, 5, 6, 7, 8, 9, 10],
+    selectedNumb: 1,
+  };
+
+  showList = () => {
+    this.setState({ count: 1 });
+  };
+
+  changeLength = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    this.setState({ selectedNumb: Number(event.target.value) });
+  };
+
+  reverseList = () => {
+    this.setState(state => ({ isReversed: !state.isReversed }));
+  };
+
+  reset = () => {
+    this.setState({
+      isReversed: false,
+      sortBy: '',
+      selectedNumb: 1,
+    });
+  };
+
+  sortAlphabetically = () => {
+    this.setState({ sortBy: 'alpha' });
+  };
+
+  sortByLength = () => {
+    this.setState({ sortBy: 'length' });
+  };
+
+  render() {
+    const {
+      count, goods, isReversed, sortBy, lengths, selectedNumb,
+    } = this.state;
+
+    const visibleGoods = goods.filter(good => good.replace(' ', '').length >= selectedNumb);
+
+    visibleGoods.sort((a, b) => {
+      switch (sortBy) {
+        case 'alpha':
+          return a.localeCompare(b);
+
+        case 'length':
+          return a.replace(' ', '').length - b.replace(' ', '').length;
+
+        default:
+          return 0;
+      }
+    });
+
+    if (isReversed) {
+      visibleGoods.reverse();
+    }
+
+    return (
+      <div className="App">
+        <div className="App__container">
+          <h1 className="App__title">Goods</h1>
+          <div className="App__info">
+            {count === 1 && (
+              <ul className="App__goods">
+                {visibleGoods.map(good => (
+                  <li className="App__good" key={good}>
+                    {good}
+                  </li>
+                ))}
+              </ul>
+            )}
+            <button
+              className={count === 1 ? 'disabled' : 'App__open'}
+              type="button"
+              onClick={this.showList}
+            >
+              Click here
+            </button>
+            <div className="App__sort">
+              <button
+                className={count === 1 ? 'App__button' : 'disabled'}
+                type="button"
+                onClick={this.reverseList}
+              >
+                Reverse
+              </button>
+              <button
+                className={count === 1 ? 'App__button' : 'disabled'}
+                type="button"
+                onClick={this.sortAlphabetically}
+              >
+                Sort alphabetically
+              </button>
+              <button
+                className={count === 1 ? 'App__button' : 'disabled'}
+                type="button"
+                onClick={this.reset}
+              >
+                Reset
+              </button>
+              <button
+                className={count === 1 ? 'App__button' : 'disabled'}
+                type="button"
+                onClick={this.sortByLength}
+              >
+                Sort by length
+              </button>
```

Comment:
you do not really need to create a separate CSS class for it. you can write count === 1 to a variable and then conditionally render the needed buttons